src/modules/routing/RoutingModule.ts

```diff
@@ -120,4 +131,5 @@ interface MediatorConfiguration {
   verkey: Verkey;
   invitationUrl: string;
   alias?: string;
+  transport?: Transport;
```

Comment:
I don't think we should save a transport with the mediator. We're moving more towards mediator coordination protocol in which case the flow probably looks something like this:

- Mediator creates invitation
- Mediatee receives invitation, sends request (probably HTTP)
- Mediatior sends response with DIDDoc containing both ws and http endpoint.
- We make a preference to use WS, but could also use HTTP. So it should be more dynamic I think.

Reply:
I see. The transport here is passed mainly to say what transport should be used for this step:
* Mediatee receives invitation, **sends request** (probably HTTP)

I assumed that it should be possible to do it either via HTTP or WS.

Maybe, we should select the transport channel based on the service endpoint in the invitation? I see that [0. Invitation to Connect](https://github.com/hyperledger/aries-rfcs/blob/master/features/0160-connection-protocol/README.md#0-invitation-to-connect) defines just one serviceEndpoint but [Aries RFC 0434: Out-of-Band Protocol 1.1](https://github.com/hyperledger/aries-rfcs/blob/master/features/0434-outofband/README.md#invitation-httpsdidcommorgout-of-bandverinvitation) shows an example with services as an array.

Reply:
> Maybe, we should select the transport channel based on the service endpoint in the invitation?

I think for creating the connection with the mediator it should use the one from the invitation, just like we do for any other connection. After that it should use the transport(s) as defined in the received did doc